pallets/acurast/src/lib.rs

```diff
@@ -214,11 +209,10 @@ pub mod pallet {
                 );
             }
 
-            T::AssetTransactor::lock_asset(
+            T::RewardManager::lock_reward(
                 registration.reward.clone(),
                 T::Lookup::unlookup(who.clone()),
-            )
-            .map_err(|_| Error::<T>::InvalidPayment)?;
+            )?;
```

Comment:
Make consistent: further down in `pay_reward` we translate to `Error::<T>::FailedToPay`, here we rely on the `DispatchError` being distinguishable by it's included message (I'm not sure if this message get's logged/displayed)